File: Linked List Group Reverse.java
class Solution {
    public static Node reverseKGroup(Node head, int k) {
        // code here
        Node next=null;
        Node prev=null;
        Node cur=new Node(0);
        cur.next=head;
        Node curr=head;
        Node currprev=null;
        while(curr!=null){
            int c=0;
            currprev=prev;
            prev=null;
            while(curr!=null && c<k){
            next=curr.next;
            curr.next=prev;
            prev=curr;
            curr=next;
            c++;
            }
             if(currprev==null ){
                cur.next=prev;
            }
            else if(currprev!=null){
            while(currprev.next!=null){
                currprev=currprev.next;
            }
            currprev.next=prev;
                
            }
            
          
         }
        return cur.next;
       
    }
}

File: Linked List Group Reverse.cpp
class Solution {
  public:
    Node *reverseKGroup(Node *head, int k) {
        // code here
        
        vector<int>v;
        Node*p=head;
        while(p!=NULL){
           v.push_back(p->data); 
           p=p->next;
        }
        
        int rem=v.size()%k;
        
        int i=0;
        while(i<v.size()-rem){
            
        reverse(v.begin()+i,v.begin()+i+k);
        i+=k;
        
        }
        
         
        if(i!=v.size()-1){
            reverse(v.begin()+i,v.end());
        }
        
        
       
        p=head;
        i=0;
        while(p!=NULL){
          p->data=v[i];
          i++;
          p=p->next;
        }
        return head;
        
    }
};
